src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float RESIZE_THRESHOLD = 0.75f;
+    private static final float RESIZE_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        if (putIntoTable(key, value)) {
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = calculateIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        if (size / (float) table.length >= RESIZE_THRESHOLD) {
+            Node<K, V>[] temp = table;
+            table = new Node[(int) (temp.length * RESIZE_MULTIPLIER)];
+            for (Node<K, V> node : temp) {
+                while (node != null) {
+                    putIntoTable(node.key, node.value);
+                    node = node.next;
+                }
+            }
+        }
+    }
+
+    private boolean putIntoTable(K key, V value) {
```

Comment:
The method 'putIntoTable' should be private because it's a utility method used inside this class only, and there's no need to expose it as part of the class's public API.